core/src/main/scala/kafka/tools/StorageTool.scala

```diff
@@ -45,89 +45,124 @@ import scala.jdk.CollectionConverters._
 import scala.collection.mutable.ArrayBuffer
 
 object StorageTool extends Logging {
+
+  /**
+   * Executes the command according to the given arguments and returns the appropriate exit code.
+   * @param args The command line arguments
+   * @return     The exit code
+   */
+  def runMain(args: Array[String]): Int = {
+    val namespace = parseArguments(args)
+    val command = namespace.getString("command")
+    val config = parseConfig(namespace.getString("config"))
```

Comment:
We can't run this when the command is `random-uuid`, as `KafkaConfig` requires some configs to be set and we don't provide any:
```
KAFKA_CLUSTER_ID="$(bin/kafka-storage.sh random-uuid)"
Exception in thread "main" org.apache.kafka.common.config.ConfigException: Missing required configuration `zookeeper.connect` which has no default value.
	at kafka.server.KafkaConfig.validateValues(KafkaConfig.scala:1223)
	at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:1214)
	at kafka.server.KafkaConfig.<init>(KafkaConfig.scala:544)
	at kafka.tools.StorageTool$.parseConfig(StorageTool.scala:91)
	at kafka.tools.StorageTool$.runMain(StorageTool.scala:57)
	at kafka.tools.StorageTool$.main(StorageTool.scala:79)
	at kafka.tools.StorageTool.main(StorageTool.scala)
```

Reply:
You're right, thanks for pointing this out. I missed that when I simplified the `Option(...).flatMap()` config bit. I've brought that back now.